lib/rspec/matchers/built_in/be_kind_of.rb

```diff
@@ -8,19 +8,23 @@ class BeAKindOf < BaseMatcher
       private
 
         def match(expected, actual)
-          if actual_object_respond_to?(actual, :kind_of?)
-            actual.kind_of?(expected)
-          elsif actual_object_respond_to?(actual, :is_a?)
-            actual.is_a?(expected)
+          result = nil
+          if executable? { result = actual.kind_of?(expected) }
+            result
+          elsif executable? { result = actual.is_a?(expected) }
+            result
           else
             raise ::ArgumentError, "The #{matcher_name} matcher requires that " \
-                                   "the actual object responds to either #kind_of? or #is_a? methods "\
-                                   "but it responds to neigher of two methods."
+                                   "the actual object responds to either #kind_of? or #is_a? methods " \
+                                   "but a `NoMethodError` was encountered instead."
           end
         end
```

Comment:
The original use of this matcher just supported `kind_of` so I propose we go back to that with a simple rescue.

```
def match(expected, actual)
  actual.kind_of? expected
rescue NoMethodError
  raise ::ArgumentError, "The #{matcher_name} matcher requires that " \
                         "the actual object responds to #kind_of? method " \
                         "but a `NoMethodError` was encountered instead."
end
```